Migrate user registration external test

Cruikshanks opened the pull request:
This migrates the external user registration test. When we reviewed what was there we found most of it duplicated validation that had already been checked elsewhere. It could also be argued that this testing is checked in `cypress/e2e/external/sharing-access-with-new-user.cy.js` but we've retained it to be on the safe side.

What we have done is follow the pattern we used for **reset-password** where we split the validation tests from the journey test, to make what we're checking clearer.

Whilst we were doing this, we realised we should really be doing the same for the **login**. 

So, in this change, we also create a `login/` folder and move the existing `cypress/e2e/external/login-logout.cy.js` in there along with a new validation of the login page test.

andrewhick commented:
Discussed with Tim and he's happy for me to merge without detailed review - no functional change.

Cruikshanks committed:
Migrate user registration external test

This migrates the external user registration test. When we reviewed what was there we found most of it was duplicating validation that had already been checked else where. It could also be argued what this is testing is checked in `cypress/e2e/external/sharing-access-with-new-user.cy.js` but we've retained it to be on the safe side.

So, in this change we create a `login/` folder and move the existing `cypress/e2e/external/login-logout.cy.js` in there along with a new validation of the login page test.

We then separate out the journey from the rest of the validation testing. The end result though is `repos/water-abstraction-ui/cypress/integration/external/user-registration.spec.js` is migrated.

Cruikshanks committed:
Merge branch 'main' into migrate-user-registration

Cruikshanks committed:
Move login and add validation test

Cruikshanks committed:
Migrate user registration tests

Cruikshanks committed:
Merge branch 'main' into migrate-user-registration
